### ISSUE TITLE
com.darksci.pardot.api.InvalidRequestException

### ISSUE THREAD

**Issue by tfranchiDD** [Crim/pardot-java-client#45]
Hi,

when we call this method with a prospect that not exists in Pardot (email): 
pardot.prospectRead(request);

we retrieve this error:
com.darksci.pardot.api.InvalidRequestException: Invalid API key or user key

How we could check if a prospect exists in Pardot without errors?

**Comment by Crim** [comment]
That error indicates that your Pardot API credentials are being rejected by Pardot's API servers.
[Authentication Documentation](http://developer.pardot.com/#authentication)

Make sure you've supplied correct credentials.

**Comment by tfranchiDD** [comment]
It is not correct because after the read we make the creation and the creation works correctly while the read gives that error.

Below an example of our code:

```
ProspectReadRequest request = new  ProspectReadRequest ();
request.selectByEmail(contactSf.getEmail());

Prospect resultProspect = null;

try {
	resultProspect = pardot.prospectRead(request);
}
catch (InvalidRequestException e) {
	log.warning("InvalidRequestException: {0} - {1}", String.valueOf(e.getCause()), String.valueOf(e.getMessage()));
}

if (resultProspect == null) {

	ProspectCreateRequest prospectCreate = new ProspectCreateRequest ();

	Prospect prospect = new Prospect ();
	prospect.setEmail(contactSf.getEmail());
	prospect.setPhone(contactSf.getMobilePhone());
	prospect.setFirstName(contactSf.getFirstName());
	prospect.setLastName(contactSf.getLastName());
	prospect.setCampaignId(this.getCampaignId());

	prospectCreate.withProspect(prospect);

	try {
		resultProspect = pardot.prospectCreate(prospectCreate);
	}
	catch (InvalidRequestException e) {

		log.warning("InvalidRequestException: {0} - {1}", String.valueOf(e.getCause()), String.valueOf(e.getMessage()));
		
	}
}
```

**Comment by Crim** [comment]
Running the following to query a prospect which does not exist:

```java
final Prospect response = client.prospectRead(new ProspectReadRequest()
     .selectByEmail("does-not-exist@example.com")
);
```

for me throws the following:

```
INFO  | 2020-02-18 18:34:36 | [main] api.PardotClient (PardotClient.java:223) - Response: RestResponse{responseStr='<?xml version="1.0" encoding="UTF-8"?>
<rsp stat="fail" version="1.0">
   <err code="4">Invalid prospect email address</err>
</rsp>
', httpCode=200}

com.darksci.pardot.api.InvalidRequestException: Invalid prospect email address

	at com.darksci.pardot.api.PardotClient.submitRequest(PardotClient.java:240)
	at com.darksci.pardot.api.PardotClient.prospectRead(PardotClient.java:754)
```

Which unfortunately is just how the library works at the moment.  Likely some work could/should be put in to make error results from the API handled in a better way.

**Comment by tfranchiDD** [comment]
I have run your code in main class and I have this error:

com.darksci.pardot.api.InvalidRequestException: Invalid http response code from server: 400

but In debug mode I have check the correct response inside library:

RestResponse{responseStr='<?xml version="1.0" encoding="UTF-8"?>
<rsp stat="fail" version="1.0">
   <err code="4">Invalid prospect email address</err>
</rsp>
', httpCode=400}

but in library code there is this:

throw new InvalidRequestException("Invalid http response code from server: " + restResponse.getHttpCode(), restResponse.getHttpCode());

**Comment by tfranchiDD** [comment]
Could the session have timed out ???

After restarting the server, I haven't had this error for a short time.

**Comment by Crim** [comment]
Ah that's possible.  I'll try to mock up a test and verify the behavior.  I thought that the library should automatically re-authenticate and replay the request that failed, but there could be a bug there.

Will report back.

**Comment by Crim** [comment]
Posted a bugfix for the session renewal issue you brought up, thanks for reporting it!

I'll release and publish a new version (1.1.2) shortly.

**Comment by Crim** [comment]
Published version 1.1.2 with the fix for this.  

I'm going to close this issue, but if you continue to have problems, or run into other issues please submit an issue and I can help get it resolved.

Thanks!
Stephen
